src/plots/polar/set_convert.js

```diff
@@ -0,0 +1,169 @@
+ *   - c2g (in plot) 'data' -> 'geometric' radians (fn of ax rotation & direction)
+ *   - t2g (in updateAngularAxis) 'tick' value (in degrees) -> 'geometric' radians
+ * + for category axes:
+ *   - d2c (in calc) just like for cartesian axes
+ *   - c2g (in plot) category indices -> 'geometric' radians
+ *   - t2g (in updateAngularAxis) 'tick' value (as category indices) -> 'geometric' radians
+ *
+ * Then, 'g'eometric data is ready to be converted to (x,y).
+ */
+module.exports = function setConvert(ax, polarLayout, fullLayout) {
+    setConvertCartesian(ax, fullLayout);
+
+    switch(ax._id) {
+        case 'x':
+        case 'radialaxis':
+            setConvertRadial(ax);
+            break;
+        case 'angular':
+        case 'angularaxis':
+            setConvertAngular(ax, polarLayout);
+            break;
+    }
+};
+
+function setConvertRadial(ax) {
+    ax.setGeometry = function() {
+        var rng = ax.range;
+
+        var rFilter = rng[0] > rng[1] ?
+            function(v) { return v <= 0; } :
+            function(v) { return v >= 0; };
+
+        ax.c2g = function(v) {
+            var r = ax.c2r(v) - rng[0];
+            return rFilter(r) ? r : 0;
+        };
+
+        ax.g2c = function(v) {
+            return ax.r2c(v + rng[0]);
+        };
+    };
+}
+
+function toRadians(v, unit) {
+    return unit === 'degrees' ? deg2rad(v) : v;
+}
+
+function fromRadians(v, unit) {
+    return unit === 'degrees' ? rad2deg(v) : v;
+}
+
+function setConvertAngular(ax, polarLayout) {
+    var axType = ax.type;
+
+    if(axType === 'linear') {
+        var _d2c = ax.d2c;
+        var _c2d = ax.c2d;
+
+        ax.d2c = function(v, unit) { return toRadians(_d2c(v), unit); };
+        ax.c2d = function(v, unit) { return _c2d(fromRadians(v, unit)); };
+    }
+
+    // override makeCalcdata to handle thetaunit and special theta0/dtheta logic
+    ax.makeCalcdata = function(trace, coord) {
+        var arrayIn = trace[coord];
+        var len = trace._length;
+        var arrayOut, i;
+
+        var _d2c = function(v) { return ax.d2c(v, trace.thetaunit); };
+
+        if(arrayIn) {
+            if(Lib.isTypedArray(arrayIn) && axType === 'linear') {
+                if(len === arrayIn.length) {
+                    return arrayIn;
+                } else if(arrayIn.subarray) {
+                    return arrayIn.subarray(0, len);
+                }
+            }
+
+            arrayOut = new Array(len);
+            for(i = 0; i < len; i++) {
+                arrayOut[i] = _d2c(arrayIn[i]);
+            }
+        }
+        return arrayOut;
+    };
+
+    // N.B. we mock the axis 'range' here
+    ax.setGeometry = function() {
+        var sector = polarLayout.sector;
+        var dir = {clockwise: -1, counterclockwise: 1}[ax.direction];
+        var rot = deg2rad(ax.rotation);
+
+        ax.rad2g = function(v) { return dir * v + rot; };
+        ax.g2rad = function(v) { return (v - rot) / dir; };
```

Comment:
As discussed in slack, seems like we can 🔪 `ax.rad2*` and `ax.*2rad` entirely now, so we don't need to think about the difference between `rad` and `g` to understand all this.

Reply:
done inhttps://github.com/plotly/plotly.js/pull/2895/commits/f5bceda77844160879033d283f2a4cf914924a7a